crates/external_services/src/grpc_client/dynamic_routing/elimination_rate_client.rs

```diff
@@ -102,13 +126,30 @@ impl EliminationBasedRouting for EliminationAnalyserClient<Client> {
             })
             .collect::<Vec<_>>();
 
-        let request = tonic::Request::new(UpdateEliminationBucketRequest {
+        let mut request = tonic::Request::new(UpdateEliminationBucketRequest {
             id,
             params,
             labels_with_bucket_name,
             config,
         });
 
+        headers
+            .tenant_id
+            .parse()
+            .map(|tenant_id| request.metadata_mut().append("x-tenant-id", tenant_id))
+            .inspect_err(|err| logger::warn!(header_parse_error=?err,"invalid x-tenant-id received in update_elimination_bucket_config"))
+            .ok();
+
+        headers.request_id.map(|request_id| {
+            request_id
+                .parse()
+                .map(|request_id| request.metadata_mut().append("x-request-id", request_id))
+                .inspect_err(|err| {
+                    logger::warn!(header_parse_error=?err,"invalid x-request-id received in update_elimination_bucket_config")
+                })
+                .ok();
+        });
+
```

Comment:
Seems like We're adding this logic in every function with a tonic request construction.
Have we tried out using a single function, something like `add_headers(headers, &mut Request<T>)`
@Chethan-rao